### ISSUE TITLE
Extension assumes puppet is installed on C drive

### ISSUE THREAD

**Issue by bloodmc** [lingua-pupuli/puppet-vscode#348]
### What Versions are you running?

OS Version: Windows 10
VSCode Version: 1.25.1
Puppet Extension Version: 0.11.0
Puppet Agent Version: 5.5.4
PDK Version (if applicable): 1.6.1.0

### What You Are Seeing?
After loading VSCode and clicking on my test pp file, I get the following error message

`Could not find a valid Puppet installation at 'C:\Program Files\Puppet Labs\Puppet\puppet'. While syntax highlighting and grammar detection will still work, intellisense and other advanced features will not.`

### What is Expected?
To locate the proper puppet installed location which can be found in registry at `HKLM\Software\Puppet Labs\Puppet\RememberedInstallDir64`

### How Did You Get This To Happen? (Steps to Reproduce)
* Install puppet agent on any drive but C.
* Launch VSCode
* Open any .pp file.
* VSCode should prompt with message.

**Comment by glennsarti** [comment]
Hrmm ok.  It seems we assume it's installed in `Program Files` but the end result is the same.  It's not where we assume it is :-)

https://github.com/lingua-pupuli/puppet-vscode/blob/master/src/configuration.ts#L55-L70

**Comment by jpogran** [comment]
@bloodmc thanks for the report. Our configuration classes make the assumptions that they do if you do not set the `puppet.puppetAgentDir` setting. If you set your custom install path for the Puppet-Agent in the `puppet.puppetAgentDir` setting then the extension will load.

You may be wondering why we're making you set the custom path if we already knew that registry key exists. It was an explicit choice on my part. VSCode is written in node and TypeScript, so we would have to use node packages to access the registry. There isn't an official Windows registry package, so we would have to choose between several options, all with different implementations and approaches. The key feature we need is the ability to read either the 32bit or 64bit key, which is accessed differently depending on the process that started it. We didn't have the time at the inception of the development to figure out which package did that correctly, without requiring elevated permissions to read registry keys.

In short, it's something we're aware of but haven't set a high priority on since there is a way to tell us where the installation is through the settings. We'll add this to the backlog and re-evaluate it after some of the higher priority issues.

**Comment by jpogran** [comment]
Since this issue was first submitted we've improved our path detection to work so long as either the PDK or Puppet-Agent was installed in a default location *or* the `installDirectory` was set. In PR #503 we set the PDK to default, which is the preferred source by most Puppet Users. In PR #513 we're implementing a setting which will automatically attempt to use PDK in the default location, then the Puppet-Agent if the PDK is not found. If either is installed in a non-default location, you can still specify `installDirectory` at either the global or workspace level and it will use that location.

With the above functionality, is registry detection still needed?
